### PR TITLE
fix: external-dns does not work when adding ip for my pihole

### LINKED ISSUE #3: bug: external dns does not work when i tried to use it add my pihole's ip resolv
The expected as followings:

```shell
2024-09-01T23:18:11+08:00 time="2024-09-01T15:18:11Z" level=info msg="Instantiating new Kubernetes client"
2024-09-01T23:18:11+08:00 time="2024-09-01T15:18:11Z" level=info msg="Using inCluster-config based on serviceaccount-token"
2024-09-01T23:18:11+08:00 time="2024-09-01T15:18:11Z" level=info msg="Created Kubernetes client https://10.43.0.1:443"
2024-09-01T23:18:12+08:00 time="2024-09-01T15:18:12Z" level=info msg="All records are already up to date"
2024-09-01T23:19:12+08:00 time="2024-09-01T15:19:12Z" level=info msg="All records are already up to date"
2024-09-01T23:20:13+08:00 time="2024-09-01T15:20:13Z" level=info msg="add pihole.k8s.local.cheverjohn.me IN A -> 192.168.31.192"
2024-09-01T23:20:13+08:00 time="2024-09-01T15:20:13Z" level=warning msg="Skipping unsupported endpoint pihole.k8s.local.cheverjohn.me TXT \"heritage=external-dns,external-dns/owner=homelab,external-dns/resource=ingress/pihole-system/pihole\""
2024-09-01T23:20:13+08:00 time="2024-09-01T15:20:13Z" level=warning msg="Skipping unsupported endpoint a-pihole.k8s.local.cheverjohn.me TXT \"heritage=external-dns,external-dns/owner=homelab,external-dns/resource=ingress/pihole-system/pihole\""
2024-09-01T23:21:13+08:00 time="2024-09-01T15:21:13Z" level=info msg="All records are already up to date"
2024-09-01T23:22:14+08:00 time="2024-09-01T15:22:14Z" level=info msg="All records are already up to date"
2024-09-01T23:23:14+08:00 time="2024-09-01T15:23:14Z" level=info msg="All records are already up to date"
```

now the error as followings:

```shell
2024-09-01T23:18:11+08:00 time="2024-09-01T15:18:11Z" level=info msg="Instantiating new Kubernetes client"
2024-09-01T23:18:11+08:00 time="2024-09-01T15:18:11Z" level=info msg="Using inCluster-config based on serviceaccount-token"
2024-09-01T23:18:11+08:00 time="2024-09-01T15:18:11Z" level=info msg="Created Kubernetes client https://10.43.0.1:443"
2024-09-01T23:18:12+08:00 time="2024-09-01T15:18:12Z" level=info msg="All records are already up to date"
2024-09-01T23:19:12+08:00 time="2024-09-01T15:19:12Z" level=info msg="All records are already up to date"
2024-09-01T23:21:13+08:00 time="2024-09-01T15:21:13Z" level=info msg="All records are already up to date"
2024-09-01T23:22:14+08:00 time="2024-09-01T15:22:14Z" level=info msg="All records are already up to date"
2024-09-01T23:23:14+08:00 time="2024-09-01T15:23:14Z" level=info msg="All records are already up to date"
2024-09-01T23:19:12+08:00 time="2024-09-01T15:19:12Z" level=info msg="All records are already up to date"
2024-09-01T23:21:13+08:00 time="2024-09-01T15:21:13Z" level=info msg="All records are already up to date"
2024-09-01T23:22:14+08:00 time="2024-09-01T15:22:14Z" level=info msg="All records are already up to date"
2024-09-01T23:23:14+08:00 time="2024-09-01T15:23:14Z" level=info msg="All records are already up to date"
2024-09-01T23:19:12+08:00 time="2024-09-01T15:19:12Z" level=info msg="All records are already up to date"
2024-09-01T23:21:13+08:00 time="2024-09-01T15:21:13Z" level=info msg="All records are already up to date"
2024-09-01T23:22:14+08:00 time="2024-09-01T15:22:14Z" level=info msg="All records are already up to date"
2024-09-01T23:23:14+08:00 time="2024-09-01T15:23:14Z" level=info msg="All records are already up to date"
2024-09-01T23:19:12+08:00 time="2024-09-01T15:19:12Z" level=info msg="All records are already up to date"
2024-09-01T23:21:13+08:00 time="2024-09-01T15:21:13Z" level=info msg="All records are already up to date"
2024-09-01T23:22:14+08:00 time="2024-09-01T15:22:14Z" level=info msg="All records are already up to date"
2024-09-01T23:23:14+08:00 time="2024-09-01T15:23:14Z" level=info msg="All records are already up to date"
2024-09-01T23:19:12+08:00 time="2024-09-01T15:19:12Z" level=info msg="All records are already up to date"
2024-09-01T23:21:13+08:00 time="2024-09-01T15:21:13Z" level=info msg="All records are already up to date"
2024-09-01T23:22:14+08:00 time="2024-09-01T15:22:14Z" level=info msg="All records are already up to date"
2024-09-01T23:23:14+08:00 time="2024-09-01T15:23:14Z" level=info msg="All records are already up to date"
```

No Add records, there are just "All records are already up to date".

### PR DESCRIPTION
**Pull request by Chever-John**
Just as the title says, this PR solves this issue.

Just add a nginx annotation rule.

Closes #3

### CODE REVIEW

**Comment by dueyfinster** [comment]
@Chever-John I feel my PR is simpler to understand for a newbie since he showed example of it in video already. I'm no K8S expert by a long way but yours appears more technically correct to me?

**Comment by Chever-John** [comment]
@dueyfinster 
# TLDR: I agree with you:)




```yaml
ingress:
  enabled: true
  hosts:
    - "pihole.home"
  annotations:
    kubernetes.io/ingress.class: "nginx-internal"
```

**Advantages**

*Flexibility*: By specifying`ingress.class` using annotations, it is easy to switch between different Ingress Controllers without modifying Ingress resources. this is useful for scenarios where you need to use different Ingress Controllers in different environments.

*Compatibility*: Many existing tools and plugins support configuring Ingress via annotations, which makes this approach more compatible in some cases.

**Disadvantages**

*Readability*: The use of annotations may result in less readable configuration files, especially if there are more annotations.

*Standardization*: The use of annotations may lead to inconsistency in configuration, especially in teamwork where different developers may use different annotations to implement the same functionality.

```yaml
ingress:
  ingressClassName: nginx-internal
  enabled: true
  hosts:
    - "pihole.home"
```

**Advantages**

*Simplicity*: Using the `ingressClassName` field directly to specify the Ingress Controller, the configuration is more concise and clear, easy to understand and maintain.

*Consistency*: By using Kubernetes native fields, it ensures consistency and standardization of the configuration and reduces configuration inconsistencies caused by different annotations.

**Disadvantages**

*Flexibility*: Compared to the approach of using annotations, using `ingressClassName` directly may not be flexible enough in some cases, especially if you need to switch Ingress Controllers dynamically.

*Support*: some older versions of Kubernetes or specific tools may not support the `ingressClassName` field.

**Comment by dueyfinster** [comment]
Thanks for the explanation - every day is a school day as they say 😄

**Comment by Chever-John** [comment]
@dueyfinster Learning never stops! 😄
